projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -1288,7 +1302,24 @@ void CN0411_cmd_solution(uint8_t *args, struct cn0411_device *cn0411_dev)
 		cn0411_dev->solution.tds_factor = TDS_NACL;
 		printf("Solution set to NaCl.\n");
 	} else {
-		printf("Incorrect input value!\n");
+		index = 0;
+		str = strtok (arg, ",");
```

Comment:
as note: be aware that `strtok()` will change the contents of `arg` to give you separate strings;
this means that the contents of `uart_rx_buffer` will be changed here ;
i can't tell if this can cause problems for this code, but it's one thing to keep an eye for

Reply:
there are two different variables `args` and `arg`. The string function is applied on `arg` which is of local scope.

Reply:
ah ; my bad